endpoints/openrtb2/auction.go

```diff
@@ -1565,8 +1565,15 @@ func setSiteImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrapper) {
 				if r.Site == nil {
 					r.Site = &openrtb2.Site{}
 				}
+				if r.Site.Publisher == nil {
+					r.Site.Publisher = &openrtb2.Publisher{}
+				}
+
 				if r.Site.Domain == "" {
 					r.Site.Domain = domain
+					r.Site.Publisher.Domain = domain
```

Comment:
At this point, is it not possible that r.Site.Publisher.Domain is populated, and getting overwritten here? If for example, r.Site.Domain is "", but r.Site.Publisher.Domain has a value, you should be able to reach this line and overwrite the existing value.